### ISSUE TITLE
Error processing Ajax request for text search

### ISSUE THREAD

**Issue by pgt74** [Wunderbyte-GmbH/moodle-local_wunderbyte_table#68]
Hello there,
Issue with text search:
Works fine on development environment (running Maria DB)
Fails on Prod (running MySQL) - while the rest of functionality works fine in Prod.
I cannot run debugger in Prod :)
The only error message I can get is "dmlreadexception" (in Ajax response, which comes with "error": true).
I captured the queries in Dev, ran them with Adminer in Prod... all of them work fine.
Any idea ?

**Comment by pgt74** [comment]
BTW, missed to say : am trying with the demo page of wunderbyte, nothing else !

**Comment by dasistwas** [comment]
Could you provide detailed steps how to reproduce the problem? @pgt74

**Comment by pgt74** [comment]
Hi David !
Found the issue eventually (not so easy, because I do not have too much
debug details on Prod platform).
In setup_fulltextsearch, if $CFG->version < 2022112800 (which is the case
of my Prod instance), you prepare the string to be searched on each record
with COALESCE(CAST(" . $value . " AS VARCHAR), ' ').;. which does not work
with MySQL :(
I replaced it with CAST AS CHAR and now everything runs fine.
Was tricky because my Dev instance runs a slightly newer version of Moodle,
so the query was built with sql_cast_to_char which uses CONCAT, running
fine on MySQL :)

OK, issue solved. Hope this experience can help somebody else :)
Kind regards.
And congrats for the great table plugin !!

Paul
